Source/santad/SNTNotificationQueue.m

```diff
@@ -69,20 +69,45 @@ - (void)addEvent:(SNTStoredEvent *)event
   @synchronized(self.pendingNotifications) {
     [self.pendingNotifications addObject:d];
   }
-  [self flushQueue];
+  [self flushQueueWithReplies:(reply != nil)];
 }
 
-- (void)flushQueue {
+- (void)flushQueueWithReplies:(BOOL)hasReplies {
```

Comment:
If a machine is configured to use standalone mode, I think this will result in no notifications being queued when a user is logged out. Is that what we want? We might instead want to perform the `reply`, and add the notification back so that a user can be informed when logging back in.

Also might want to consider if this `hasReplies` is necessary. We likely want to flush (or reply and re-add) entries any time there isn't an XPC connection, not just when trying to add a new event.
I think this could maybe be simplified. It shouldn't matter if the current event has a reply block. I think we should always remove notifications

Reply:
Yeah this wasn't the best. I've re-worked things to put flushQueue back to how it was before but now we only save the reply block in the dict if a notifier connection is currently available. There's still the possibility that the connection goes away after the block is saved in the dict but this should be very rare.